src/common.h

```diff
@@ -369,6 +369,24 @@ char *ly_strnchr(const char *s, int c, size_t len);
  */
 int ly_strncmp(const char *refstr, const char *str, size_t str_len);
 
+/**
+ * @brief Get all possible prefixes from an arbitrary YANG string by iteratively returning specific substrings.
+ *
+ * The function looks for possible prefix at the beginning of the @p str_begin and returns it.
+ * If there is some string at the @p str_begin and then follows the prefix, then the string is returned.
+ * If @p str_begin does not contain any prefix, then the entire @p str_begin is returned.
+ * The @p next can be used as the @p str_begin in the next iteration.
+ *
+ * @param[in] str_begin Begin of the input string.
+ * @param[in] str_end Length of the @p str_begin. If set to NULL then the @p str_begin must be NULL-terminated string.
+ * @param[out] is_prefix Type of substring found. Set to 1 for prefix, otherwise 0.
+ * @param[out] next Rest of the string after the found substring.
```

Comment:
Just to be consistent with your other parameter names, maybe call this `str_next`.

Reply:
ok